recommend/runtimePolicy.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	opb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/observability"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const serviceAccData = "/run/secrets/kubernetes.io/serviceaccount"
+
+// createRuntimePolicy function generates runtime policy for service account
+func createRuntimePolicy(img *ImageInfo) error {
+	var labels string
+	for key, value := range img.Labels {
+		labels = strings.TrimPrefix(fmt.Sprintf("%s,%s=%s", labels, key, value), ",")
+	}
+	gRPC := ""
+	if val, ok := os.LookupEnv("DISCOVERY_SERVICE"); ok {
+		gRPC = val
+	} else {
+		gRPC = "localhost:9089"
+	}
+	// create a client
+	conn, err := grpc.Dial(gRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return errors.New("could not connect to the server. Possible troubleshooting:\n- Check if discovery engine is running\n- Create a portforward to discovery engine service using\n\t\033[1mkubectl port-forward -n explorer service/knoxautopolicy --address 0.0.0.0 --address :: 9089:9089\033[0m\n[0m")
+	}
+	defer conn.Close()
+	client := opb.NewObservabilityClient(conn)
+	podResp, err := client.GetPodNames(context.Background(), &opb.Request{
+		Label:     labels,
+		NameSpace: img.Namespace,
+	})
+	if err != nil {
+		return err
+	}
+	for _, podName := range podResp.PodName {
```

Comment:
This `podName` here, is it the exact name of pod? If yes, then consider following scenario..

In the `observability.Summary()`, if we get the following data..
```
deployment_1 --> pod_1 | service_account_file --> accessed by --> binary_1
deployment_1 --> pod_2 | service_account_file --> accessed by --> binary_2
```
Since for each pod we're create a new `MatchSpec` and converting it to a policy, this would create two `MatchSpec` and since the `MatchSpec.Name` is the same, policy file name will be the same and last policy written to the disk will overwrite the first policy.

@vishnusomank Let me if this could happen.

Reply:
Yes, the `podName` is the exact name of the pod. And the scenario mentioned might happen since we are not differentiating them based on pod name. Ideally, the scenario is highly improbable, right? If I have 2 pods (ie rs =2) then their behavior also should be the same.
Should we add `runtime-PODNAME` as part of the policy name to avoid this scenario? @wazir-ahmed

Reply:
@vishnusomank No, we should not create multiple policy file. There should only be one policy file per deployment for service account access. What we should do is.. we should combine the `fromSource` information from all the pods (of a particular deployment) and create a policy like below.
```yaml
file:
  matchPaths:
  - path: "/run/secrets/kubernetes.io/serviceaccount"
    fromSource:
    - path: <binary_1>
    - path: <binary_2>
    - path: <binary_n>
```
@nyrahul This is the expected output, right?